### PR TITLE
added hw-24-1

### PR DESCRIPTION
**Pull request by kenu21**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java [hunk 1]
```diff
@@ -24,10 +24,11 @@ public BucketDaoJdbcImpl(Connection connection) {
 
     @Override
     public Bucket create(Bucket bucket) {
-        try (Statement statementBuckets = connection.createStatement()) {
-            String queryBucktes = String.format("INSERT INTO buckets (user_id) VALUES (%d);",
-                    bucket.getUserId());
-            statementBuckets.executeUpdate(queryBucktes, Statement.RETURN_GENERATED_KEYS);
+        String queryBucktes = "INSERT INTO buckets (user_id) VALUES ?;";
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java]:
```suggestion
        String queryBuckets = "INSERT INTO buckets (user_id) VALUES ?";
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java [hunk 2]
```diff
@@ -41,10 +42,10 @@ public Bucket create(Bucket bucket) {
     @Override
     public Bucket get(Long bucketId) {
         Bucket bucket = null;
-        try (Statement statementBuckets = connection.createStatement()) {
-            String queryBucktes = String.format("SELECT * from buckets where bucket_id = %d;",
-                    bucketId);
-            ResultSet resultSet = statementBuckets.executeQuery(queryBucktes);
+        String queryBucktes = "SELECT * FROM buckets WHERE bucket_id = ?;";
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java]:
```suggestion
        String queryBuckets = "SELECT * FROM buckets WHERE bucket_id = ?";
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java [hunk 3]
```diff
@@ -110,11 +113,12 @@ public Boolean addItem(Long bucketId, Long itemId) {
...
-            statementBuckets.executeUpdate(queryBucktes);
+        String queryBucktes =
+                "DELETE FROM buckets_items WHERE bucket_id = ? AND item_id = ?;";
+        try (PreparedStatement statementBuckets = connection.prepareStatement(queryBucktes)) {
+            statementBuckets.setLong(1, bucketId);
+            statementBuckets.setLong(2, itemId);
+            statementBuckets.executeUpdate();
         } catch (SQLException e) {
             logger.error("Can't delete item", e);
         }
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java:126]: is it correct to return `true` everywhere?
maybe better return true in try block and return false in catch block?

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java [hunk 4]
```diff
@@ -101,22 +106,24 @@ public Order delete(Long id) {
...
                 order.setId(orderId);
                 List<Item> listItem = new ArrayList<>();
-                try (Statement statementItems = connection.createStatement()) {
-                    String queryItems = String.format(
-                            "SELECT * FROM items i INNER JOIN orders_items oi "
-                            + "ON i.item_id = oi.item_id "
-                                    + "INNER JOIN orders o ON oi.order_id = o.order_id "
-                            + "WHERE o.user_id = %d and o.order_id = %d;", userId, orderId);
-                    ResultSet resultSetItems = statementItems.executeQuery(queryItems);
+                String queryItems =
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java]: maybe better divide this method into two parts? second part will have name getAllItemsFromOrder(Long orderId, Long userId)?

**Review by Sofasmile — COMMENTED** [review]
Good, but make small fixes)

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java (hunk 2 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java (hunk 3 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java:126]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java (hunk 4 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java (hunk 3 above)
Comment (Sofasmile) [reply src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java:126]: the same in other methods:)

**Review by Sofasmile — APPROVED** [review]

File: src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java (hunk 3 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java:126]: OK:)

**Pull request merged**